pkg/upgrade/upgrader/jiva_upgrade.go

```diff
@@ -633,3 +706,203 @@ func jivaUpgrade(pvName, openebsNamespace string, utaskObj *utask.UpgradeTask) (
 	klog.Info("Upgrade Successful for", pvName)
 	return options.utaskObj, nil
 }
+
+func (j *jivaVolumeOptions) cleanup(openebsNamespace string) error {
+	var err error
+	if j.replicaObj.version == currentVersion && j.replicaObj.name != "" {
+		klog.Info("cleaning old replica deployment")
+		err = deployClient.WithNamespace(j.ns).Delete(j.replicaObj.name, &metav1.DeleteOptions{})
+		if err != nil {
```

Comment:
handle isNotFound error as not error

Reply:
If the job restarts and the replica deployment is missing the `getReplica` function will not populate `replicaObj.name` and the check `j.replicaObj.name != ""` will skip the deletion.